specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/networkManagerEffectiveConfiguration.json

```diff
@@ -0,0 +1,180 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "NetworkManagerEffectiveConfiguration",
+    "description": "The Microsoft Azure Network Manager effective configuration API provides a RESTful set of web services that interact with Microsoft Azure Networks service to manage your network resources. The API has entities that capture the relationship between an end user and the Microsoft Azure Networks service.",
+    "version": "2020-08-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworks/{virtualNetworkName}/networkManagerEffectiveConfigurations": {
+      "parameters": [
+        {
+          "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+        },
+        {
+          "$ref": "#/parameters/ResourceGroupNameParameter"
+        },
+        {
+          "$ref": "#/parameters/VirtualNetworkNameParameter"
+        },
+        {
+          "$ref": "./network.json#/parameters/ApiVersionParameter"
+        },
+        {
+          "$ref": "#/parameters/ListPageSizeParameter"
+        },
+        {
+          "$ref": "#/parameters/ListSkipTokenParameter"
+        }
+      ],
+      "get": {
```

Comment:
I see a GET API for networkManagerEffectiveConfigurations. Where is the corresponding PUT API?

Reply:
networkManagerEffectiveConfigurations  is not a nested resource. And may I ask why Post is better than Get here? since when we talked with ARM team, they preferred GET than POST.

Reply:
1. Resource payload doesn't represent a list of resources (proxy/tracked), it is not as per ARM guidelines. 
2. Refer https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc004-uri-is-not-following-arm-standard-guidelines
3. Please share if you have any communication around this.

Reply:
Does 004 mean Get must be used with Delete/Put?

Reply:
Had an internal conversation, GET is recommended pattern for this kind of API and so you do not need to change here. The general recommendation is to model resources returned in GET response to follow ARM resource model - id, name, type defined.